td/client.py

```diff
@@ -557,10 +599,10 @@ def _make_request(self, method: str, endpoint: str, mode: str = None, params: di
         """
 
         url = self._api_endpoint(endpoint=endpoint)
-        headers = self._headers(mode=mode)
 
         # Make sure the token is valid if it's not a Token API call.
         self.validate_token()
+        headers = self._headers(mode=mode)

```

Comment:
Pretty sure token validation should happen BEFORE setting the headers for the request, otherwise you're potentially still sending the expired access token from prior to obtaining the new one, right?